libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs

```diff
@@ -72,6 +73,16 @@ type GalleyApi =
     :<|> FedEndpoint "send-mls-commit-bundle" MessageSendRequest MLSMessageResponse
     :<|> FedEndpoint "query-group-info" GetGroupInfoRequest GetGroupInfoResponse
     :<|> FedEndpoint "on-client-removed" ClientRemovedRequest EmptyResponse
+    :<|> FedEndpoint "on-typing-indicator-updated" TypingDataUpdateRequest EmptyResponse
+
+data TypingDataUpdateRequest = TypingDataUpdateRequest
+  { tdurTypingData :: TypingData,
+    tdurConnection :: ConnId,
```

Comment:
You should not pass a connection ID as part of a federated request. Rather the function which is taking care of sending events should accept an optional connection ID, because there won't be one when the request originates from a remote backend.

Reply:
I wasn't sure if this was needed over federation, thanks for clarifying. Removed / made optional as applicable.